### PR TITLE
Wizard recipe: samtools-v1.10.0

### PR DESCRIPTION
**Pull request by amckay1**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: samtools
* Version: v1.10.0

@staticfloat please review and merge.

### CODE REVIEW

File: S/samtools/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "samtools"
+version = v"1.10.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/samtools/samtools.git", "f01547ee339b68629f754702ba678a63121fbdfa"),
+    DirectorySource("./bundled")
```
Comment (giordano) [inline S/samtools/build_tarballs.jl:11]: The patch is empty
```suggestion
```

File: S/samtools/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+cd samtools/
```
Comment (giordano) [inline S/samtools/build_tarballs.jl:20]:
```suggestion
cd $WORKSPACE/srcdir/samtools/
```

File: S/samtools/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+autoheader
+autoconf -Wno-syntax
+      ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make
+make install
+exit
```
Comment (giordano) [inline S/samtools/build_tarballs.jl:26]: Judging by the [`README.md`](https://github.com/samtools/samtools/blob/07ad69a6aea7d8b4f40d6a7d73e4c97c5d50f0ef/README.md#building-samtools) this should use something like `--with-htslib=${libdir}` to actually make use of our htslib
```suggestion
./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-htslib=${libdir}
make -j${nproc}
make install
```

File: S/samtools/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("x86_64", "macos"; )
+]
```
Comment (giordano) [inline S/samtools/build_tarballs.jl:38]: Can we build for all platforms and see why they fail?
```suggestion
platforms = supported_platforms(; experimental=true)
```

File: S/samtools/bundled/patches/samtools.patch [hunk 5]
```diff
@@ -0,0 +1,3 @@
+diff --git before/samtools/.git/index after/samtools/.git/index
```
Comment (giordano) [inline S/samtools/bundled/patches/samtools.patch:1]: This file can be deleted, it's empty

**Review by giordano — CHANGES_REQUESTED** [review]
Thanks for your contribution!

**Comment by amckay1** [comment]
Thanks for reviewing! Still learning about BinaryBuilder.jl but these changes look good to me :)

File: S/samtools/build_tarballs.jl (hunk 3 above)
Comment (amckay1) [reply S/samtools/build_tarballs.jl:26]: I was wondering about this, and I considered specifying something like `/workspace/x86_64-linux-gnu-libgfortran3-cxx03/destdir/include/htslib/include`, but then the build proceeded without needing to specify so long as I added in the htslib jll file. Do you think we still need to specify `${libdir}`?

Comment (giordano) [reply S/samtools/build_tarballs.jl]: Setting `CPPFLAGS` was enough to find everything.

File: S/samtools/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "samtools"
+version = v"1.10.0"
```
Comment (giordano) [inline S/samtools/build_tarballs.jl:6]: Any particular reasons for using this version and not [v1.14](https://github.com/samtools/samtools/releases/tag/1.14) which is the latest one?

Comment (amckay1) [reply S/samtools/build_tarballs.jl:6]: I tried to use v1.14 but I hit an error building that seemed related to htslib_jll that I was using (which is at 1.10). I tried downgrading to the same version and that seemed to fix the issue.

I'm happy to try and make a htslib 1.14 in the future and then an equivalent samtools, but I thought I better get my feet wet first :)

Comment (giordano) [reply S/samtools/build_tarballs.jl:6]: > I'm happy to try and make a htslib 1.14 in the future 

Too late, I did it a few hours ago: #3879 🙂 what do you want to do? Merge this as is or upgrade to v1.14?

Comment (amckay1) [reply S/samtools/build_tarballs.jl:6]: Awesome, I'm all for upgrading to v1.14, would that require a new pull request or can we specify the new version here?

Comment (giordano) [reply S/samtools/build_tarballs.jl:6]: You can do it here

Comment (giordano) [reply S/samtools/build_tarballs.jl]: Ok, I did it for you :slightly_smiling_face:

**Review by giordano — APPROVED** [review]
Thanks for sticking with us!

**Comment by amckay1** [comment]
Thanks for taking this over the finish line! Got caught up with work but I see it was a straightforward upgrade to v1.14. Will try to provide a more refined recipe next time! :)

**Pull request merged**